website/versioned_docs/version-experimental/RelayHooks-StepByStep.md

```diff
@@ -0,0 +1,300 @@
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      query: text,
+      variables,
+    }),
+  });
+
+  // Get the response as JSON
+  return await response.json();
+}
+
+export default fetchGraphQL;
+```
+
+### 2.3: Fetching GraphQL From React
+
+Now we can use our `fetchGraphQL` function to fetch some data in our app. Open `src/App.js` and edit it as follows:
+
+```javascript
+// your-app-name/src/App.js
+import React from 'react';
+import './App.css';
+import fetchGraphQL from './fetchGraphQL';
+
+const { useState, useEffect } = React;
+
+function App() {
+  // We'll load the name of a repository, initially setting it to null
+  const [name, setName] = useState(null);
+
+  // When the component mounts we'll fetch a repository name
+  useEffect(() => {
+    let isMounted = true;
+    fetchGraphQL(`
+      query RepositoryNameQuery {
+        # feel free to change owner/name here
+        repository(owner: "facebook" name: "relay") {
+          name
+        }
+      }
+    `).then(response => {
+      // Avoid updating state if the component unmounted before the fetch completes
+      if (!isMounted) {
+        return;
+      }
+      const data = response.data;
+      setName(data.repository.name);
+    }).catch(error => {
+      console.error(error);
+    });
+
+    return () => {
+      isMounted = false;
+    };
+  }, [fetchGraphQL]);
+
+  // Render "Loading" until the query completes
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>
+          {name != null ? `Repository: ${name}` : "Loading"}
+        </p>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+## Step 3: When To Use Relay
+
+At this point we can fetch data with GraphQL and render it with React. This is a reasonable solution that can be sufficient for many apps. However, this approach doesn't necessarily scale. As our app grows in size and complexity, or the number of people working on the app grows, a simple approach like this can become limiting. Relay provides a number of features designed to help keep applications fast and reliable even as they grow in size and complexity: colocating data dependencies in components with GraphQL fragments, data consistency, mutations, etc. Check out [Thinking in GraphQL](/docs/en/thinking-in-graphql.html) and [Thinking in Relay](/docs/en/thinking-in-relay.html) for an overview of how Relay makes it easier to work with GraphQL.
+
+
+## Step 4: Adding Relay To Our Project
+
+In this guide we'll demonstrate installing the *experimental* release of Relay Hooks, a new, hooks-based Relay API that supports React Concurrent Mode and Suspense. 
+
+First we'll add the necessary packages. Note that Relay is comprised of three key pieces: a compiler (which is used at build time), a core runtime (that is React agnostic), and a React integration layer.
+
+```bash
+# NPM Users
+npm install --save relay-runtime react-relay@experimental
```

Comment:
we don't explicitly need to install relay-runtime since it's a dependency of react-relay, right? might make this simpler

Reply:
4.2 explicitly imports relay-runtime, though, so it's a little odd not to explicitly specify that as a dependency.

if it's feasible from an OSS perspective, it might be reasonable API-wise to add a simple helper to set up a relay environment to `react-relay` that e.g. just takes in the fetch function, and otherwise makes so that users don't need to touch both `react-relay` and `relay-runtime`.

Reply:
@taion that's exactly what we were thinking, having a `createEnvironment(fetch)` helper